testsuite/openshift/objects/auth_config/__init__.py

```diff
@@ -50,7 +50,7 @@ def create_instance(cls, openshift: OpenShiftClient, name, route: Route,
                 "namespace": openshift.project
             },
             "spec": {
-                "hosts": hostnames or route.hostnames
+                "hosts": hostnames or (route.hostnames if route else None) or []
```

Comment:
why this change?

Reply:
there will be an error if hostnames and route are None. Because NoneType has no attributes

Reply:
But you need either hostnames or route, otherwise it doesnt make any sense

Reply:
yes, that's just an error treatment. there is an `add_host` function below to add hosts manually if you need it

Reply:
I would add that only after we found usecase where that makes sense, I think for now the current way is alright